File: Java/Day02Assignment/MyOwnAutoShopProject/Truck.java
package com.sonata;

public class Truck extends Car {
int weight;

public Truck(int a,double b, String c){
	super(a,b,c);
}
	public double getSalePrice() {
		double price;
		if(weight>2000) {
			 price = regularPrice - (regularPrice*0.1);
		}
		else {
			 price = regularPrice - (regularPrice*0.2);
		}
		return price;
	}
public void display() {
	System.out.println("Truck : ");
	System.out.println("weight : " + weight);
	System.out.println("speed : " + speed);
	System.out.println("color : " + color);
	System.out.println("original Price : " + regularPrice);
	System.out.println("SalePrice : " + getSalePrice());
	System.out.println();
}
public static void main(String arg[]) {
	
}
}

File: Typescripts/Truck.ts
import { Car } from "./Car";

class Truck extends Car{

    weight:Number;

    getSalePrice(): void {

        if(this.weight >2000){
            var pc = this.regularPrice - (this.regularPrice*.1);
            console.log(pc);
        }

        else{
            var pc = this.regularPrice - (this.regularPrice*.2);
            console.log(pc);
        }

    }

}
var t1 = new Truck();
t1.weight=3000;
t1.regularPrice=4000;
t1.speed = 80;
t1.color= "Orange";
t1.getSalePrice();